src/components/layout/PrimaryLayout/PrimaryLayout.js

```diff
@@ -22,32 +24,38 @@ export function PrimaryLayout ({
   showSideMenu = true,
   showAlphaFlag = true
 }) {
+  const [isHposConnectionAlive, setIsHposConnectionAlive] = useState()
   const { setIsConnected, isConnected } = useConnectionContext()
   const { setCurrentUser } = useCurrentUserContext()
+  const { newMessage } = useFlashMessageContext()
+  const { push } = useHistory()
 
   const onError = ({ graphQLErrors }) => {
     const { isHposConnectionActive } = graphQLErrors
-    setIsConnected(isHposConnectionActive)
-  }
-
-  const onCompleted = ({ hposSettings }) => {
-    if (hposSettings) setIsConnected(true)
+    if (!isHposConnectionActive) {
+      setIsHposConnectionAlive(false)
+    } else {
+      setIsHposConnectionAlive(isHposConnectionActive)
+    }
   }
 
-  const { data: { hposSettings: settings = {} } = {} } = useQuery(HposSettingsQuery, { pollInterval: 10000, onCompleted, onError, notifyOnNetworkStatusChange: true, ssr: false })
-  const { newMessage } = useFlashMessageContext()
+  const { data: { hposSettings: settings = {} } = {} } = useQuery(HposSettingsQuery, { pollInterval: 10000, onError, notifyOnNetworkStatusChange: true, ssr: false })
 
   useEffect(() => {
-    if (!isConnected) {
+    setIsConnected(isHposConnectionAlive && wsConnection)
+    if (isConnected === false) {
       newMessage('Your Holoport is currently unreachable.', 0)
+      if (window.location.pathname !== '/' && window.location.pathname !== '/admin/login') {
```

Comment:
would be good to have this in a utility function in `utils/urls`, just so we can localize knowledge of what paths mean what. 
(I don't think we're doing anything like that very consistently yet, but we should be moving towards it)
-- this comment is not blocking.